addons/l10n_in/models/res_partner.py

```diff
@@ -77,6 +77,7 @@ def _onchange_country_id(self):
     @api.onchange('vat')
     def onchange_vat(self):
         if self.vat and self.check_vat_in(self.vat):
+            self.vat = self.vat.upper()
```

Comment:
We already depend on base_vat where it should be converted to a standard format.  So, I do not think the onchange itself should do it.

Reply:
May be we can define new method in `base_vat` i.e. `format_vat_in`. Since we don't have any specific for India we only have `check_vat_in`

Reply:
If necessary, but I wonder if stdnum would not already handle it.

Reply:
https://github.com/odoo/odoo/blob/d997f388c392af503238d813963d085d111d7938/addons/base_vat/models/res_partner.py#L104

@jco-odoo I believe the issue lies in the `_split_vat()` method. This method splits the VAT number and checks the country code based on the first characters. Many countries use the first letters of the VAT to identify the country. However, In India the first two characters represent the state rather than the country.

https://github.com/odoo/odoo/blob/d997f388c392af503238d813963d085d111d7938/addons/base_vat/models/res_partner.py#L671

Here as a result, when the method checks the first characters of the VAT against the country code, it fails and returns the unformatted VAT instead of formatting it.

Reply:
I am improving in https://github.com/odoo/odoo/pull/179078

Reply:
We can continue with this for the moment.